pkg/validation/limits.go

```diff
@@ -50,6 +50,9 @@ type Limits struct {
 	MaxProfileStacktraceDepth        int `yaml:"max_profile_stacktrace_depth" json:"max_profile_stacktrace_depth"`
 	MaxProfileSymbolValueLength      int `yaml:"max_profile_symbol_value_length" json:"max_profile_symbol_value_length"`
 
+	// Distributor per-app usage breakdown.
+	DistributorUsageGroups []map[string]string `yaml:"distributor_usage_groups" json:"distributor_usage_groups"`
```

Comment:
Do you mind explaining why this is a slice of maps, rather than just a map, which I would expect?

Here the yaml "explanation of my question":

```yaml
current_usage_groups:
- my_team: '{namespace="cool-stuff"'
- other_team: '{namespace="boring-stuff"}'

why_not_usage_groups:
 my_team: '{namespace="cool-stuff"}'
 other_team: '{namespace="boring-stuff"}'	
```

Reply:
Totally get the order and deterministic result problem (now  that you mentioned it😆 ), but I think the map inside the slice will still be non-deterministic in the current implementation.

I think maybe a slice of two string might be a better solution:

```[]struct{
name string
selector string
}
```

or just sorting by the map key (name of the rule group).

Probably it is best to aim for consistency with other Grafana Products and do what Mimir does.